lib/carrierwave/storage/fog.rb

```diff
@@ -1,6 +1,10 @@
 # encoding: utf-8
 
-require "fog" unless defined?(::Fog)
+begin
+  require "fog/core"
+rescue LoadError
+  require "fog" unless defined?(::Fog)
```

Comment:
After `require "fog/core"`, there *is* a `Fog` defined. (see https://github.com/fog/fog-core/blob/master/lib/fog/core.rb and https://github.com/fog/fog-core/blob/master/lib/fog/core/version.rb)